app/controllers/application_controller.rb

```diff
@@ -82,24 +82,20 @@ def set_theme
     end
   end
 
-  def not_found
-    raise ActionController::RoutingError, 'User Not Found'
-  end
-
   def find_user
     login = params[:login] || params[:user_id] || params[:id]
     @user = User.where(login: login).first || (current_user if %w[new favorites].include? action_name)
-    not_found unless @user
+    user_not_found_message unless @user
```

Comment:
@sudara noticed new method definition here. Should I remove the old ones?

Reply:
`user_not_found_message` just sets @message for the 404 page — `not_found` actually raises a routing error so we 404.... so I think this change isn't desirable.

Reply:
> so I think this change isn't desirable.

I don't remember what exactly I meant here :) But `not_found` is no longer defined from the looks of it.

Reply:
It's in application_controller, as it's used by more than this controller.

Reply:
Done. Reverted this change and I see now what was happening :)